### DISCUSSION TITLE
JioFiber APIs

### DISCUSSION THREAD

**Discussion by itsyourap** [JFC-Group/JF-Customisation#20] [General]
These are the APIs used in the JioFiber which enables apps like JioHome to communicate with the router.

There are two types of APIs defined in the JioFiber Router:-

- `http://192.168.29.1/GET_CGI` - Can be used to get information from the Router
- `http://192.168.29.1/SET_CGI` - Can be used to alter information and settings in the Router

The files in `/pfrm2.0/share/lua/5.1/teamf1lualib/easyMesh****.lua` (from the router firmware) have the necessary information regarding the APIs.

I am attaching a Postman collection which has most of the APIs along with their responses in different cases.

Note that all of the requests need to be POST requests and all of them expects a JSON POST data. They also produce a response in JSON format. 

Also note that most of the requests will require a cookie header with a cookie named `uid` which is derived on login or forced login. All of the requests are provided in the collection.

### Download : [JioFiber.postman_collection.zip](https://github.com/itsyourap/JioFiber-Home-Gateway/files/10349847/JioFiber.postman_collection.zip)

Download the attached file and extract it. You will get a JSON file which consists of a Postman collection.
I have well documented most of the parts of requests and responses via comments.

**Please delete the comments in the POST Data before sending any request otherwise you will get an `500 Internal Server Error` message.**

To import the collection, follow [this tutorial](https://learning.postman.com/docs/getting-started/importing-and-exporting-data/).

After importing the collection, please refer to the collection variables and alter them as needed. Change the value of `adminName` and `adminPassword` as per your router credentials. The value of `uid` variable changes upon execution due to the Pre-Request Script. Disable it if you need.

Also, the last request in the collection named `Customer Details` is a GET request that **is sent to Jio's server and NOT TO YOUR ROUTER** which brings back information regarding the JioFiber connection (customer name, details etc etc) and an JWT auth token which is used by apps like JioTV, etc etc to login and get subscriptions.

**Comment by A-viral-dev** [comment]
I am getting error "JSONError: "undefined" is not valid JSON".
1. I changed collection variables as per my local.
2. I removed all comments in login Json.

**Comment by A-viral-dev** [comment]
I am not getting this error if i remove UID part in pre-script.
If I modify pre-script and run "Login" Now I am getting new error "Error: Blocked IPv4 address".

**Reply by itsyourap** [reply]
Are you on Postman website? If yes, then please download Postman App and run the requests locally.

**Comment by f0rr0** [comment]
@itsyourap Were you able to figure out what endpoints MyJio app uses for fetching wifi online / active devices?
The web interface uses https://www.jio.com/api/jio-device-management-service/resource/default-device/ONLINE for example

**Reply by f0rr0** [reply]
The iOS SSL cert seems to be pinned on the MyJio app so I am unable to inspect the traffic

**Comment by SR7007** [comment]
Can we set the pon mode of JF4032 to EPON or Xpon, i tried and after reboot it revert to gpon.

**Reply by shailparmar** [reply]
How did you changed the mode from gpon to epon?
